crates/cargo-platform/src/lib.rs

```diff
@@ -88,15 +88,26 @@ impl Platform {
                             )),
                         _ => (),
                     },
-                    Cfg::KeyPair(name, _) => if name.as_str() == "feature" {
-                        warnings.push(String::from(
-                            "Found `feature = ...` in `target.'cfg(...)'.dependencies`. \
-                             This key is not supported for selecting dependencies \
-                             and will not work as expected. \
-                             Use the [features] section instead: \
-                             https://doc.rust-lang.org/cargo/reference/features.html"
-                        ))
+                    Cfg::KeyPair(name, _) => match name.as_str() {
+                        "feature" => {
+                            warnings.push(String::from(
+                                "Found `feature = ...` in `target.'cfg(...)'.dependencies`. \
+                                 This key is not supported for selecting dependencies \
+                                 and will not work as expected. \
+                                 Use the [features] section instead: \
+                                 https://doc.rust-lang.org/cargo/reference/features.html"
+                            ));
+                        },
+                        "version" => {
+                            warnings.push(String::from(
+                                "Found `version = ...` in `target.'cfg(...)'.dependencies`. \
+                                This format is ignored by Cargo. \
+                                Use the format `version(\"1.23.4\")` or `version(\"1.23\")` instead."
+                            ));
+                        },
```

Comment:
Based on https://github.com/rust-lang/cargo/pull/15533#discussion_r2093685172 I don't think this should be present.